cmake/modules/FindDEAL_II_PETSC.cmake

```diff
@@ -193,6 +193,17 @@ if(NOT PETSC_PETSCVARIABLES MATCHES "-NOTFOUND")
 
     endif()
   endforeach()
+
+  # PETSc does not expose Kokkos version, so we need to search for Kokkos
+  # ourselves.
+  if(PETSC_WITH_KOKKOS)
+    file(STRINGS "${PETSC_PETSCVARIABLES}" KOKKOS_INCLUDE
+      REGEX "^KOKKOS_INCLUDE =.*")
+    string(REGEX REPLACE "^KOKKOS_INCLUDE = -I" "" KOKKOS_INCLUDE "${KOKKOS_INCLUDE}")
+    find_package(Kokkos 3.7.0 QUIET
+      HINTS ${KOKKOS_INCLUDE}/..
+      )
+  endif()
```

Comment:
do we need to fail here is we don't find kokkos or find the bundled version?

Reply:
I can switch from `QUIET` to `REQUIRED` if you prefer. `find_package` won't find the bundled version, I've tested it.

Reply:
Don't use `REQUIRED` - You need to set PETSC to not found instead.

Another improvement would be to disallow finding Kokkos outside of the path provided as hint.

Reply:
Done